com.ibm.wala.cast.js/src/main/java/com/ibm/wala/cast/js/callgraph/fieldbased/WorklistBasedOptimisticCallgraphBuilder.java

```diff
@@ -86,60 +93,114 @@ public Set<Pair<CallVertex, FuncVertex>> extractCallGraphEdges(
       }
     }
 
-    while (!worklist.isEmpty()) {
-      MonitorUtil.throwExceptionIfCanceled(monitor);
-
-      Vertex v = worklist.iterator().next();
-      worklist.remove(v);
-      Set<FuncVertex> vReach = MapUtil.findOrCreateSet(reachingFunctions, v);
-      for (Vertex w : Iterator2Iterable.make(flowgraph.getSucc(v))) {
+    if (bound < 0) {
```

Comment:
Having a separate case for `bound < 0` leads to a bunch of code duplication (again hindering maintainability).  Why not just use the `else` case of this `if`, but write `while (bound == -1 || cnt < bound)` as the loop condition?  I think it's ok for all the pending call stuff to run even in the unbounded case.